pseudomonas_analysis/2_identify_generic_genes_pathways_pao1.ipynb

```diff
@@ -0,0 +1,1764 @@
+    "        )\n",
+    "\n",
+    "        # Read in metadata\n",
+    "        metadata = pd.read_csv(mapping_file, header=0, sep=\"\\t\", index_col=0)\n",
+    "\n",
+    "        selected_metadata = metadata.loc[experiment_id]\n",
+    "        sample_ids = list(selected_metadata[sample_id_colname])\n",
+    "\n",
+    "    return sample_ids\n",
+    "\n",
+    "def shift_template_experiment_tmp(\n",
+    "    normalized_data_file,\n",
+    "    selected_experiment_id,\n",
+    "    sample_id_colname,\n",
+    "    NN_architecture,\n",
+    "    dataset_name,\n",
+    "    scaler,\n",
+    "    local_dir,\n",
+    "    base_dir,\n",
+    "    run,\n",
+    "):\n",
+    "    \"\"\"\n",
+    "    Generate new simulated experiment using the selected_experiment_id as a template\n",
+    "    experiment using the same workflow as `simulate_by_latent_transform`\n",
+    "\n",
+    "    This will return a file with a single simulated experiment following the workflow mentioned.\n",
+    "    This function can be run multiple times to generate multiple simulated experiments from a\n",
+    "    single selected_experiment_id.\n",
+    "\n",
+    "    Arguments\n",
+    "    ----------\n",
+    "    normalized_data_file: str\n",
+    "        File containing normalized gene expression data\n",
+    "\n",
+    "        ------------------------------| PA0001 | PA0002 |...\n",
+    "        05_PA14000-4-2_5-10-07_S2.CEL | 0.8533 | 0.7252 |...\n",
+    "        54375-4-05.CEL                | 0.7789 | 0.7678 |...\n",
+    "        ...                           | ...    | ...    |...\n",
+    "\n",
+    "    selected_experiment_id: str\n",
+    "        Experiment id selected as template\n",
+    "\n",
+    "    sample_id_colname: str\n",
+    "        Column header that contains sample id that maps expression data and metadata\n",
+    "\n",
+    "    NN_architecture: str\n",
+    "        Name of neural network architecture to use.\n",
+    "        Format 'NN_<intermediate layer>_<latent layer>'\n",
+    "\n",
+    "    dataset_name: str\n",
+    "        Name for analysis directory. Either \"Human\" or \"Pseudomonas\"\n",
+    "\n",
+    "    scaler: minmax model\n",
+    "        Model used to transform data into a different range\n",
+    "\n",
+    "    local_dir: str\n",
+    "        Parent directory on local machine to store intermediate results\n",
+    "\n",
+    "    base_dir: str\n",
+    "        Root directory containing analysis subdirectories\n",
+    "\n",
+    "    run: int\n",
+    "        Simulation run\n",
+    "\n",
+    "    Returns\n",
+    "    --------\n",
+    "    simulated_data_file: str\n",
+    "        File containing simulated gene expression data\n",
+    "\n",
+    "    \"\"\"\n",
+    "\n",
+    "    # Files\n",
+    "    NN_dir = os.path.join(base_dir, dataset_name, \"models\", NN_architecture)\n",
+    "    latent_dim = NN_architecture.split(\"_\")[-2] # Here we manually changed the index to work\n",
```

Comment:
I don't know how frequently you use the `NN_architecture` string, but you may want to just pass the latent dimension as a parameter rather than having this function's behavior be dependent on the architecture name

Reply:
Ha...you found the source of the problem. For some reason past me thought it made sense to extract the latent dim from the `NN_architecture` string. I have added an issue to ponyo to fix this in a new version: https://github.com/greenelab/ponyo/issues/26

For now I'm doing it this hacky way because this repo is mostly using functionality found in an earlier version of ponyo. So to avoid more things breaking and having to update ponyo and this repo in parallel I've frozen the version of ponyo used here. Not an ideal solution. But something I chose to do to keep the analysis of this repo moving forward in a timely manner.